app/src/org/commcare/CommCareApplication.java

```diff
@@ -1476,4 +1478,7 @@ public boolean checkPendingBuildRefresh() {
         return false;
     }
 
+    public PrototypeFactory getPrototypeFactory(Context c) {

```

Comment:
Interesting - and you cant override static methods I'm guessing?

Having to get the singleton CommCareApplication to call this method that doesn't actually need the instance seems wrong to me. Am I missing something or is this just a tradeoff we have to accept to have tests?